staging/src/k8s.io/client-go/discovery/cached/disk/diskv_cache.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package disk
+
+// NOTE(negz): This file is copied from the upstream httpcache implementation,
+// which is no longer maintained. It has been altered to prevent each cache
+// write being fsynced.
+// https://github.com/gregjones/httpcache/blob/901d90/diskcache/diskcache.go
```

Comment:
:warning:  The original file is MIT licensed whereas the file preamble says Apache 2.

Reply:
Ah, good catch.